scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,530 @@
+#
+# Copyright © 2013-2024 NVIDIA CORPORATION & AFFILIATES. ALL RIGHTS RESERVED.
+#
+# This software product is a proprietary product of Nvidia Corporation and its affiliates
+# (the "Company") and all right, title, and interest in and to the software
+# product, including all associated intellectual property rights, are and
+# shall remain exclusively with the Company.
+#
+# This software product is governed by the End User License Agreement
+# provided with the software product.
+#
+
+import json
+import re
+import signal
+import subprocess
+import argparse
+import logging
+import sys
+from typing import List
+
+
+def configure_logger():
+    logger_name = "standby_node_checker"
+    logger = logging.getLogger(logger_name)
+
+    if not logger.hasHandlers():
+        logger.setLevel(logging.INFO)
+
+        handler = logging.StreamHandler()
+
+        formatter = logging.Formatter(
+            fmt="%(asctime)s %(levelname)s: %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
+        )
+        handler.setFormatter(formatter)
+
+        logger.addHandler(handler)
+
+    return logger
+
+
+logger = configure_logger()
+
+
+def set_log_level(log_level):
+    numeric_level = getattr(logging, log_level.upper(), logging.INFO)
+    logger.setLevel(numeric_level)
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Checking if standby node is ready.")
+
+    parser.add_argument(
+        "--fabric-interfaces",
+        nargs="+",
+        required=True,
+        help="Specify one or more fabric interfaces (at least one is required), eg ib0, mlx3_0",
+    )
+
+    parser.add_argument(
+        "--mgmt-interface",
+        required=True,
+        help="Specify one management interfaces (at least one is required), eg eth0",
+    )
+
+    parser.add_argument(
+        "--log-level",
+        type=str,
+        choices=["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"],
+        default="INFO",
+        help="Set the logging level.",
+    )
+    parser.add_argument("--version", action="version", version="%(prog)s 1.0.0")
+
+    args = parser.parse_args()
+    return args
+
+
+class StandbyNodeHealthChecker:
+    UFM_HA_CLUSTER_COMMAND = "ufm_ha_cluster {}"
+    SYSTEMCTL_IS_ACTIVE_COMMAND = "systemctl is-active {}.service"
+    IS_HA_COMMAND = UFM_HA_CLUSTER_COMMAND.format("is-ha")
+    IS_MASTER_COMMAND = UFM_HA_CLUSTER_COMMAND.format("is-master")
+    HA_STATUS_COMMAND = UFM_HA_CLUSTER_COMMAND.format("status")
+    IBDEV2NETDEV_COMMAND = "ibdev2netdev"
+    IBSTAT_COMMAND = "ibstat"
+    IP_LINK_SHOW_COMMAND = "ip --json link show"
```

Comment:
are you sure the --json option works on all platforms? on my RHEL7 it doesn't work?